pkg/users/account-create-dialog.js

```diff
@@ -283,28 +377,48 @@ export function account_create_dialog(accounts) {
         const props = {
             id: "accounts-create-dialog",
             title: _("Create new account"),
-            body: <AccountCreateBody state={state} errors={errors} change={change} />
         };
 
+        if (state.dialogLoading) {
+            props.body = (
+                <Bullseye>
+                    <Spinner isSVG />
+                </Bullseye>
+            );
+        } else {
+            props.body = <AccountCreateBody state={state} errors={errors} change={change} shells={shells} />;
+        }
+
         const footer = {
             actions: [
                 {
                     caption: _("Create"),
                     style: "primary",
                     clicked: () => {
-                        return passwd_check(false, state.real_name, state.user_name, state.password, state.password_confirm, state.locked, state.change_passw_force);
+                        return passwd_check(false, false, state.real_name, state.user_name, state.password, state.password_confirm, state.locked, state.home_dir, state.change_passw_force);
                     },
-                    disabled: state.confirm_weak
+                    disabled: state.confirm_weak || state.home_dir_exists
                 }
             ]
         };
+        if (state.home_dir_exists) {
+            footer.actions.push(
+                {
+                    caption: _("Create and change ownership of home directory"),
+                    style: "warning",
+                    clicked: () => {
+                        return passwd_check(false, true, state.real_name, state.user_name, state.password, state.password_confirm, state.locked, state.home_dir);
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230314-140716-6b50d398-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)